File: T1.SqlSharp/T1.SqlSharp/Expressions/UnaryOperatorExtensions.cs
namespace T1.SqlSharp.Expressions;

public static class UnaryOperatorExtensions
{
    public static string ToSql(this UnaryOperator op)
    {
        return op switch
        {
            UnaryOperator.BitwiseNot => "~",
            UnaryOperator.Not => "NOT",
            _ => throw new ArgumentOutOfRangeException(nameof(op), op, null)
        };
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/UnaryOperatorExtensions.ts
import { UnaryOperator } from './UnaryOperator';

export function unaryOperatorToSql(op: UnaryOperator): string {
    switch (op) {
        case UnaryOperator.BitwiseNot: return '~';
        case UnaryOperator.Not: return 'NOT';
        default: throw new Error('Unknown UnaryOperator: ' + op);
    }
} 